src/libsyntax/parse/parser.rs

```diff
@@ -4437,6 +4438,38 @@ impl<'a> Parser<'a> {
         })
     }
 
+    /// Parses the following grammar:
+    ///     TraitItemAssocTy = Ident ["<"...">"] [":" [TyParamBounds]] ["where" ...] ["=" Ty]
+    fn parse_trait_item_assoc_ty(&mut self, preceding_attrs: Vec<Attribute>)
```

Comment:
@nikomatsakis Is there a reason why `parse_trait_item_assoc_ty` doesn't consume the semicolon? I think it really could given that it's only used in one place and it wouldn't make any sense to parse all of this without the semicolon at the end. The only invocation of this function immediately expects a semicolon right after it.

(added this question again because GitHub removed it on the latest diff)

Reply:
It doesn't make a big difference either way. A lot of times it's good to leave separators and terminators out from functions like this, since it makes them more re-usable, but I think in this case the `;` logically "belongs" to the trait item, so it would make sense for this function to consume it.

Reply:
In the latest code, the semicolon is consumed.